File: src/types/task.ts
export type TaskPriority = 'high' | 'medium' | 'low'

export type TaskStatus = 'pending' | 'completed'

export type TaskSchedulingStatus = 'unscheduled' | 'scheduled'

export type Task = {
  id: string
  title: string
  status: TaskStatus
  priority: TaskPriority
  deadlineDate?: string
  deadlineTime?: string
  estimatedDurationMinutes?: number
  categoryId?: string
  schedulingStatus: TaskSchedulingStatus
  scheduledEventId: string | null
  scheduledAt: string | null
  createdAt: string
  updatedAt: string
}

export type TaskInput = Omit<
  Task,
  | 'id'
  | 'schedulingStatus'
  | 'scheduledEventId'
  | 'scheduledAt'
  | 'createdAt'
  | 'updatedAt'
>

File: backend/src/models/Task.cpp
#include "models/Task.h"

namespace voicecalendar::models
{

std::string toStorageValue(TaskStatus status)
{
    return status == TaskStatus::Completed ? "completed" : "pending";
}

std::optional<TaskStatus> taskStatusFromStorageValue(const std::string& value)
{
    if (value == "pending")
    {
        return TaskStatus::Pending;
    }
    if (value == "completed")
    {
        return TaskStatus::Completed;
    }
    return std::nullopt;
}

std::string toStorageValue(TaskPriority priority)
{
    switch (priority)
    {
    case TaskPriority::High:
        return "high";
    case TaskPriority::Medium:
        return "medium";
    case TaskPriority::Low:
        return "low";
    }
    return "medium";
}

std::optional<TaskPriority> taskPriorityFromStorageValue(const std::string& value)
{
    if (value == "high")
    {
        return TaskPriority::High;
    }
    if (value == "medium")
    {
        return TaskPriority::Medium;
    }
    if (value == "low")
    {
        return TaskPriority::Low;
    }
    return std::nullopt;
}

} // namespace voicecalendar::models
